packages/@aws-cdk/aws-cognito/lib/user-pool-attr.ts

```diff
@@ -0,0 +1,254 @@
+   * and an ISO 3166-1 Alpha-2 [ISO3166‑1] country code in uppercase, separated by a dash.
+   * For example, en-US or fr-CA.
+   */
+  LOCALE = 'locale',
+
+  /**
+   * Middle name of the user.
+   */
+  MIDDLE_NAME = 'middle_name',
+
+  /**
+   * User's full name in displayable form including all name parts, possibly including titles and suffixes.
+   */
+  NAME = 'name',
+
+  /**
+   * Casual name of the user that may or may not be the same as the given_name.
+   */
+  NICKNAME = 'nickname',
+
+  /**
+   * User's preferred telephone number.
+   *
+   * Phone numbers must follow these formatting rules: A phone number must start with a plus (+) sign, followed
+   * immediately by the country code. A phone number can only contain the + sign and digits. You must remove any other
+   * characters from a phone number, such as parentheses, spaces, or dashes (-) before submitting the value to the
+   * service.
+   */
+  PHONE_NUMBER = 'phone_number',
+
+  /**
+   * URL of the user's profile picture.
+   * This URL must refer to an image file (for example, a PNG, JPEG, or GIF image file).
+   */
+  PICTURE = 'picture',
+
+  /**
+   * Shorthand name by which the user wishes to be referred to.
+   */
+  PREFERRED_USERNAME = 'preferred_username',
+
+  /**
+   * URL of the user's profile page.
+   */
+  PROFILE = 'profile',
+
+  /**
+   * The user's time zone
+   */
+  TIMEZONE = 'zoneinfo',
+
+  /**
+   * Time the user's information was last updated.
+   * Its value is a JSON number representing the number of seconds from 1970-01-01T0:0:0Z
+   * as measured in UTC until the date/time.
+   */
+  UPDATED_AT = 'updated_at',
+
+  /**
+   * URL of the user's web page or blog.
+   */
+  WEBSITE = 'website'
+}
+
+/**
+ * Represents a custom attribute type.
+ */
+export interface ICustomAttribute {
+  /**
+   * Bind this custom attribute type to the values as expected by CloudFormation
+   */
+  bind(): CustomAttributeConfig;
+}
+
+/**
+ * Configuration that will be fed into CloudFormation for any custom attribute type.
+ */
+export interface CustomAttributeConfig {
+  // tslint:disable:max-line-length
+  /**
+   * The data type of the custom attribute.
+   *
+   * @see https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_SchemaAttributeType.html#CognitoUserPools-Type-SchemaAttributeType-AttributeDataType
+   */
+  readonly dataType: string;
+  // tslint:enable:max-line-length
+
+  /**
+   * The constraints attached to this custom attribute.
+   * The structure here would be the fragment of `CfnUserPool.SchemaAttributeProperty` associated with this data type.
+   * For example, in the case of the 'String' data type, this would be `{ "stringAttributeConstraints": { "minLength": "..", "maxLength": ".." } }`.
+   * @see https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_SchemaAttributeType.html
+   * @default - no constraints
+   */
+  readonly constraints?: { [key: string]: any };
```

Comment:
L1 is leaking here (`any`?)... Why? Can't we provide a better experience?

Reply:
You are correct. I was being lazy. It's fixed up now.